config/database.go

```diff
@@ -75,12 +78,18 @@ func NewDatabaseStore(dsn string) (ds *DatabaseStore, err error) {
 //
 // Uses MEDIUMTEXT on MySQL, and TEXT on sane databases.
 func initializeConfigurationsTable(db *sqlx.DB) error {
+	var count int
+	err := db.Get(&count, `SELECT COUNT(*) FROM `+migrationsTableName+` WHERE Name = 'create_configurations'`)
+	if err == nil && count != 0 {
+		return nil
+	}
+
 	mysqlCharset := ""
 	if db.DriverName() == "mysql" {
 		mysqlCharset = "DEFAULT CHARACTER SET utf8mb4"
 	}
 
-	_, err := db.Exec(`
+	_, err = db.Exec(`
 		CREATE TABLE IF NOT EXISTS Configurations (
```

Comment:
This is duplicating the migration code in 2 places. Can we add a comment here saying that any changes to this and the other table needs to be replicated in the migration files as well?

Reply:
Actually what do you think of adding a separate migrations table for configuration? And thinking about a case where configuration database and mattermost database can be separated, it doesn't make sense to run migrations from server store.

Reply:
I agree, and I think some customers indeed have separate databases. So in this case, I was thinking to directly import morph and use the same morph init+up workflow like the store. We'll have 2 separate migrations (config+store) running, but they are backed by different migration tables which makes sense.